creator-node/src/ImageProcessingQueue.js

```diff
@@ -0,0 +1,94 @@
+const Bull = require('bull')
+const os = require('os')
+const config = require('./config')
+const { logger: genericLogger } = require('./logging')
+
+const imageProcessingMaxConcurrency = config.get('imageProcessingMaxConcurrency')
+
+const PROCESS_NAMES = Object.freeze({
+  resizeImage: 'resizeImage'
+})
+
+// Maximum concurrency set to config var if provided
+// Otherwise, uses the number of CPU cores available to node
+const MAX_CONCURRENCY = imageProcessingMaxConcurrency !== -1
+  ? imageProcessingMaxConcurrency
+  : os.cpus().length
+
+class ImageProcessingQueue {
+  constructor () {
+    this.queue = new Bull(
+      'image-processing-queue',
+      {
+        redis: {
+          port: config.get('redisPort'),
+          host: config.get('redisHost')
+        }
+      }
+    )
+
+    this.queue.process(
+      PROCESS_NAMES.resizeImage,
+      MAX_CONCURRENCY,
+      `${__dirname}/resizeImage.js`
```

Comment:
This is what's happening here correct?
https://www.npmjs.com/package/bull#separate-processes

Reply:
yep! very intentional